### PR TITLE
Fix typing attributes and add more tests

### PR DESCRIPTION
**Pull request by rnystrom**
Fixes GitHawkApp/GitHawk#2365

### CODE REVIEW

File: Examples/Examples/ViewController.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ class ViewController: MessageViewController, UITableViewDataSource, UITableViewD
         borderColor = .lightGray
 
         messageView.textViewInset = UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 16)
-        messageView.font = UIFont.preferredFont(forTextStyle: .body)
```
Comment (BasThomas) [inline Examples/Examples/ViewController.swift]: No big deal, but why did you change this?

File: MessageViewController/DictionaryString+NSAttributedStringKey.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//
+//  Created by Ryan Nystrom on 10/29/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+extension Dictionary where Key == String {
+    
+    var attributed: [NSAttributedStringKey: Any] {
```
Comment (BasThomas) [inline MessageViewController/DictionaryString+NSAttributedStringKey.swift]: Neat 👌

File: MessageViewController/MessageTextView.swift [hunk 3]
```diff
@@ -15,30 +15,48 @@ public protocol MessageTextViewListener: class {
...
+    }
 
-    open var defaultTextColor: UIColor? = .black
+    open var defaultTextColor = UIColor.black {
+        didSet {
+            defaultTextAttributes[NSAttributedStringKey.foregroundColor.rawValue] = defaultTextColor
+        }
+    }
+
+    internal var defaultTextAttributes: [String: Any] = {
```
Comment (BasThomas) [inline MessageViewController/MessageTextView.swift]: Why is this a `[String: Any]` rather than an `[NSAttributedStringKey: Any]` now?

File: Examples/Examples/ViewController.swift (hunk 1 above)
Comment (rnystrom) [reply Examples/Examples/ViewController.swift]: Realized this is already the default

File: MessageViewController/MessageTextView.swift (hunk 3 above)
Comment (rnystrom) [reply MessageViewController/MessageTextView.swift]: Matching `typingAttributes: [String: Any]`. There's something wrong w/ `typingAttributes`'s type: [see docs](https://developer.apple.com/documentation/uikit/uitextview/1618629-typingattributes)
